miasm2/ir/translators/z3_ir.py

```diff
@@ -0,0 +1,308 @@
+import logging
+import operator
+
+import z3
+
+import miasm2.expression.expression as m2_expr
+from miasm2.ir.translators.translator import Translator
+
+log = logging.getLogger("translator_z3")
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(logging.Formatter("%(levelname)-5s: %(message)s"))
+log.addHandler(console_handler)
+log.setLevel(logging.WARNING)
+
+class Z3Mem(object):
```

Comment:
It's a good thing that two accesses to a same address with different sizes return two different arrays.

But it would be interesting to abstract memory in a way which allow: @32[EAX] == @16[EAX] . @16[EAX + 2].
It seems to be a frequent case, especially in cryptography's algorithms (a nice Z3 use case).

A solution could be a memory abstraction using an array of bytes, a DWORD access becoming a 4 elements subarray.
Unfortunately, I'm not enough familiar with Z3 to offer you a better idea...